app/javascript/components/public_health/query/AdvancedFilter.js

```diff
@@ -237,6 +237,41 @@ class AdvancedFilter extends React.Component {
     }));
   };
 
+  /**
+   * Adds another multi filter statement at certain index
+   * Ensures no field is repeated when a new statement is added
+   * @param {Object} filter - Current multi filter object containing all the possible fields
+   * @param {Number} statementIndex - Current overall statement filter index
+   */
+  addMultiStatement = (filter, statementIndex) => {
+    const currentMultiFilter = this.state.activeFilterOptions[parseInt(statementIndex)];
+    let possibleFields = currentMultiFilter.filterOption.fields;
+    currentMultiFilter.value.forEach(value => {
+      possibleFields = possibleFields.filter(field => field.name !== value.name);
+    });
+    const newValue = [...currentMultiFilter.value, this.getDefaultMultiValue(filter, possibleFields[0].name)];
+    this.changeValue(statementIndex, newValue);
+  };
+
+  /**
+   * Removes one of the multi filter statements at certain index
+   * @param {Number} statementIndex - Current overall statement filter index
+   * @param {Number} multiIndex - Index of the statement within the multi filter that will be removed
+   */
+  removeMultiStatement = (statementIndex, multiIndex) => {
+    const currentMultiFilter = this.state.activeFilterOptions[parseInt(statementIndex)];
+    let oldValue = [...currentMultiFilter.value];
+
+    // if removing the last multi statement, remove the whole filter
+    // otherwise just remove that multi statement
+    if (oldValue.length === 1) {
+      this.removeStatement(statementIndex);
+    } else {
+      const newValue = oldValue.slice(0, multiIndex).concat(oldValue.slice(multiIndex + 1, oldValue.length));
```

Comment:
Is this just removing an element out of an array? There should be a better way with `splice`
https://stackoverflow.com/questions/5767325/how-can-i-remove-a-specific-item-from-an-array
```
const array = [2, 5, 9];

const index = array.indexOf(5);
if (index > -1) {
  array.splice(index, 1);
}

// array = [2, 9]

Reply:
fixed here: https://github.com/SaraAlert/SaraAlert/pull/821/commits/0beb57efa303876fa3aea5c08ba570a5967b80ce